arm/ui/json_api.py

```diff
@@ -415,38 +415,43 @@ def abandon_job(job_id):
         db.session.commit()
         return json_return
 
+    # Kill the process id
     try:
-        # Get job id
         job = Job.query.get(int(job_id))
-        # Stop job process id (pid)
         job.status = JobState.FAILURE.value
-        job_process = psutil.Process(job.pid)
-        job_process.terminate()  # or p.kill()
-        # release job from drive
-        job.drive.release_current_job()
-        notification = Notifications(f"Job: {job_id} was Abandoned!",
-                                     f'Job with id: {job_id} was successfully abandoned. No files were deleted!')
-        db.session.add(notification)
-        db.session.commit()
-        json_return['success'] = True
-        app.logger.debug(f"Job {job_id} was abandoned successfully")
-        job.eject()
-    except psutil.NoSuchProcess:
-        db.session.rollback()
-        json_return['Error'] = f"Couldn't find job.pid - {job.pid}! Reverting db changes."
-        app.logger.debug(f"Couldn't find job.pid - {job.pid}! Reverting db changes.")
-    except psutil.AccessDenied:
-        db.session.rollback()
-        json_return['Error'] = f"Access denied abandoning job: {job.pid}! Reverting db changes."
-        app.logger.debug(f"Access denied abandoning job: {job.pid}! Reverting db changes.")
+        try:
+            job_process = psutil.Process(job.pid)
+            job_process.terminate()  # or p.kill()
+        except psutil.NoSuchProcess:
+            message = f"Process id {job.pid} was not found. Job has already been terminated."
+            app.logger.warning(message)
+            # No raise here. No process is a terminated process.
+        except psutil.AccessDenied as err:
+            message = f"Access denied abandoning job: {job.pid}!"
+            app.logger.error(message)
+            raise ValueError(message) from err
+        else:
+            app.logger.debug(f"Job {job_id} was terminated.")
     except Exception as error:
-        db.session.rollback()
-        app.logger.debug(f"Job {job_id} couldn't be abandoned. - {error}")
         json_return["Error"] = str(error)
+        json_return['success'] = False
+    else:
+        job.eject()  # only release/eject the job if the process got killed.
+        json_return['success'] = True
+
+    # Handle Errors
     if 'Error' in json_return:
-        notification = Notifications(f"Job ERROR: {job_id} couldn't be abandoned", json_return["Error"])
-        db.session.add(notification)
-        db.session.commit()
+        title = f"Job ERROR: {job_id} couldn't be abandoned."
```

Comment:
I don't see where in the code that the job is abounded and written to the database. if there is in running pid, the job should still be updated to 'failed'to prevent it reporting as active on the webpage

Reply:
I restructured the job termination into another function. This way it looks way cleaner.

Reply:
Thanks, I will have a look at the changes. Abandoning jobs has been a bug bear for a while, either failing or leaving hanging PIDs